crates/orchestrator/src/tests/jobs/da_job/mod.rs

```diff
@@ -1,64 +1,310 @@
-use crate::jobs::da_job::DaJob;
-use crate::jobs::types::{ExternalId, JobItem, JobStatus, JobType};
-use crate::jobs::Job;
+use crate::tests::common::init_config;
+
+/// Tests the DA Job's handling of a blob length exceeding the supported size.
+/// It mocks the DA client to simulate the environment and expects an error on job processing.
+/// Validates the error message for exceeding blob limits against the expected output.
+/// Asserts correct behavior by comparing the received and expected error messages.
+#[rstest]
+#[case(
+    "src/tests/jobs/da_job/test_data/state_update/640641.txt",
+    "src/tests/jobs/da_job/test_data/nonces/640641.txt",
+    "640641",
+    110
+)]
+#[case(
+    "src/tests/jobs/da_job/test_data/state_update/631861.txt",
+    "src/tests/jobs/da_job/test_data/nonces/631861.txt",
+    "631861",
+    110
+)]
+#[case(
+    "src/tests/jobs/da_job/test_data/state_update/638353.txt",
+    "src/tests/jobs/da_job/test_data/nonces/638353.txt",
+    "63853",
+    110
+)]
+#[tokio::test]
+async fn test_da_job_process_job_failure_on_impossible_blob_length(
+    #[case] state_update_file: String,
+    #[case] noances_file: String,
+    #[case] internal_id: String,
+    #[case] current_blob_length: u64,
+) -> Result<()> {
+    // Mocking DA client calls
+    let mut da_client = MockDaClient::new();
+    // dummy state will have more than 1200 bytes
+    da_client.expect_max_blob_per_txn().with().returning(|| 1);
+    da_client.expect_max_bytes_per_blob().with().returning(|| 1200);
+
+    let server = TestConfigBuilder::new().mock_da_client(Box::new(da_client)).build().await;
+    let config = config().await;
+
+    let state_update = read_state_update_from_file(state_update_file.as_str()).expect("issue while reading");
+
+    let state_update = MaybePendingStateUpdate::Update(state_update);
+    let state_update = serde_json::to_value(&state_update).unwrap();
+    let response = json!({ "id": 640641,"jsonrpc":"2.0","result": state_update });
+
+    get_nonce_attached(&server, noances_file.as_str());
+
+    let state_update_mock = server.mock(|when, then| {
+        when.path("/").body_contains("starknet_getStateUpdate");
+        then.status(200).body(serde_json::to_vec(&response).unwrap());
+    });
+
+    let max_blob_per_txn = config.da_client().max_blob_per_txn().await;
+
+    let response = DaJob
+        .process_job(
+            config.as_ref(),
+            &mut JobItem {
+                id: Uuid::default(),
+                internal_id: internal_id.to_string(),
+                job_type: JobType::DataSubmission,
+                status: JobStatus::Created,
+                external_id: ExternalId::String(internal_id.to_string().into_boxed_str()),
+                metadata: HashMap::default(),
+                version: 0,
+            },
+        )
+        .await;
+
+    match response {
+        Ok(_) => {
+            return Err(eyre!("This testcase's process_job was supposed to throw an error, it successed instead."))
```

Comment:
Valid

Reply:
we're still returning a Result

Reply:
### Figure out what is a preferred way to go while testing in Rust ? 
`Return Statements or Panic` ? 

Here's an summary of my study on what convention to follow :
- Both are used together in crates, there's no hard and fast rule as to which one to use.
Use Results :
- When you need to explicitly test the error handling logic of a function.
Use Panic for :
- When you want less boiler plate and simplicity.

I noticed arguments about using Result in functions where you need to test `async` function so that you can use the `?` operator to return what the `async` function returned, but this can also be handled using a `is_ok()` removing the need to use `?` operator completely. making this argument not important enough.

I believe we should go with `not returning anything and panicking when the test fails` on all the tests and `using results return statement` wherever explicitly required.

### Supporting Docs : 
- [Rust by Examples](https://doc.rust-lang.org/stable/rust-by-example/testing/unit_testing.html#tests-and-)
- [The Rust Book](https://doc.rust-lang.org/book/ch11-01-writing-tests.html#using-resultt-e-in-tests)
- [Using Result in main](https://doc.rust-lang.org/rust-by-example/error/result.html#using-result-in-main)

Reply:
> When you need to explicitly test the error handling logic of a function.

Why would the test case return a result for this? The function to be tested would return a Result and you would assert it in the test right?

Reply:
I believe it's for convenience.
Quoting : [The Rust Book](https://doc.rust-lang.org/book/ch11-01-writing-tests.html#using-resultt-e-in-tests:~:text=Writing%20tests%20so%20they%20return%20a%20Result%3CT%2C%20E%3E%20enables%20you%20to%20use%20the%20question%20mark%20operator%20in%20the%20body%20of%20tests%2C%20which%20can%20be%20a%20convenient%20way%20to%20write%20tests%20that%20should%20fail%20if%20any%20operation%20within%20them%20returns%20an%20Err%20variant.)
`Writing tests so they return a Result<T, E> enables you to use the question mark operator in the body of tests, which can be a convenient way to write tests that should fail if any operation within them returns an Err variant.`

Reply:
## After in-person discussion, this conversation is closed with the following decision :
### No need to return any error from the test, use `panic!` and `assert_xyz!` wherever the test needs to fail.